iwyu_test_util.py

```diff
@@ -419,6 +430,78 @@ def _GetLaunchArguments(cc_file):
   return shlex.split(args)
 
 
+class Features:
+  """ Implements feature support functions for prerequisites. """
+
+  # Parses the output of 'include-what-you-use -print-targets'.
+  SUPPORTED_TARGET_RE = re.compile(r'^\s+([a-z0-9_-]+)\s* - .*$')
+
+  def __init__(self):
+    # Maps all known features names to prerequisite predicates.
+    self.known = {
+      "target": Features.Target
```

Comment:
My concern with just `target` is that it might be confused with the compiler target triple.

Also it's not the target that IWYU would compile by default, so I feel `supports-target` would be preferable. Maybe `supports-arch`? Not sure.

Reply:
I found it reads really poorly with `IWYU_UNSUPPORTED: supports-target(arm64)`...

Reply:
Good point. Some other options - `can-target(arm64)`, `valid-target(arm64)`, `targetable(arm64)`

Reply:
Thinking about this overnight, `IWYU_UNSUPPORTED: supports-targets(arm64)` (or whatever we end up calling it) is not something anyone would ever use in a test.

If a test doesn't select a `-target` then the prerequisite would be saying that `arm64` support is somehow causing our test to fail even when it's not being used. Same if the test is running with a non-`arm64` target - support for the arm64 target would somehow be polluting the behavior of the target being used.

The final case to consider is when the test asks for `-target arm64`: the prerequisite would be saying the test is invalid if IWYU has support for target. That's clearly a contradiction.

Obviously it should be accepted syntax. If it reads wrong, I don't think that's not the worst thing - as it will make the reviewer think a bit more (hopefully). What must read well is the positive case `IWYU_REQUIRES: supports-target(arm64)`

Reply:
Yeah, good call, that makes sense.

Maybe `has-target`?

Reply:
Pushed `has-target` to see what it looks like